### PR TITLE
[fluent-bit] Add support for securityContext

### PR DESCRIPTION
**Pull request by Garnaalkroket**
In order to comply with certain audit policies I'd like to set a securityContext on the  fluent-bit pod.
It seems in the past a podSecurityPolicy was supplied, but that mechanism is deprecated now.
Would it be possible to add this support?

Kind regards,
Robin

### CODE REVIEW

File: charts/fluent-bit/templates/daemonset.yaml [hunk 1]
```diff
@@ -34,6 +34,10 @@ spec:
         {{- toYaml . | nindent 8 }}
         {{- end }}
     spec:
+      {{- if .Values.podSecurityContext }}
+      securityContext:
+      {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      {{- end}}
```
Comment (zanhsieh) [inline charts/fluent-bit/templates/daemonset.yaml:40]:
```suggestion
      {{- with .Values.podSecurityContext }}
      securityContext:
        {{- toYaml .| nindent 8 }}
      {{- end}}
```

File: charts/fluent-bit/templates/daemonset.yaml [hunk 2]
```diff
@@ -48,6 +52,10 @@ spec:
         - name: fluent-bit-loki
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
+          {{- if .Values.securityContext }}
+          securityContext:
+          {{- toYaml .Values.securityContext | nindent 12 }}
+          {{- end }}
```
Comment (zanhsieh) [inline charts/fluent-bit/templates/daemonset.yaml:58]:
```suggestion
          {{- with .Values.securityContext }}
          securityContext:
            {{- toYaml . | nindent 12 }}
          {{- end }}
```

**Review by zanhsieh — CHANGES_REQUESTED** [review]
Please bump the version in Chart.yaml. Thank you.

File: charts/fluent-bit/templates/daemonset.yaml (hunk 1 above)
Comment (Garnaalkroket) [reply charts/fluent-bit/templates/daemonset.yaml]: Thanks! Should be fixed.

**Review by zanhsieh — CHANGES_REQUESTED** [review]
Please sign DCO. Thanks. For instruction please see the bottom of this PR page `Details` next to `DCO Action required after 1s — DCO`.

**Pull request closed without merging**